src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -254,29 +254,16 @@ void PocoHTTPClient::makeRequestInternal(
             }
             LOG_DEBUG(log, "Received headers: {}", headers_ss.str());
 
-            if (status_code >= 300)
+            if (status_code == 429 || status_code == 503)
+            { // API throttling
+                ProfileEvents::increment(select_metric(S3MetricType::Throttling));
+            }
+            else if (status_code >= 300)
             {
-                String error_message;
-                Poco::StreamCopier::copyToString(response_body_stream, error_message);
-
-                if (Aws::Http::IsRetryableHttpResponseCode(response->GetResponseCode()))
-                    response->SetClientErrorType(Aws::Client::CoreErrors::NETWORK_CONNECTION);
-                else
-                    response->SetClientErrorType(Aws::Client::CoreErrors::USER_CANCELLED);
-
-                response->SetClientErrorMessage(error_message);
```

Comment:
Why we don't set `SetClientErrorMessage` and `SetClientErrorType`, but only `SetResponseBody`. How does client that reads response will get that it's an error?

Reply:
By response code. Client error is not HTTP error but rather a connection error/SSL, logical errors and so on.

Reply:
Ok, we set `response->SetResponseCode` above got it.